server/service/src/processors/transfer/shipment/update_outbound_shipment_status.rs

```diff
@@ -0,0 +1,82 @@
+use repository::{
+    InvoiceRow, InvoiceRowRepository, InvoiceRowStatus, InvoiceRowType, RepositoryError,
+    StorageConnection,
+};
+
+use crate::processors::transfer::shipment::Operation;
+
+use super::{ShipmentTransferProcessor, ShipmentTransferProcessorRecord};
+
+const DESCRIPTION: &'static str = "Update outbound shipment status from inbound shipment";
+
+pub(crate) struct UpdateOutboundShipmentStatusProcessor;
+
+impl ShipmentTransferProcessor for UpdateOutboundShipmentStatusProcessor {
+    fn get_description(&self) -> String {
+        DESCRIPTION.to_string()
+    }
+
+    /// Outbound shipment status will be updated when all below conditions are met:
```

Comment:
this approach of documenting and laying out the conditions is nice! clear and easy to follow 😁